orm/src/main/java/org/hibernate/search/hcore/impl/SearchFactoryReference.java

```diff
@@ -22,16 +20,16 @@
  */
 public final class SearchFactoryReference implements Service {
 
-	private volatile CompletableFuture<ExtendedSearchIntegrator> extendedIntegratorFuture;
+	private volatile ExtendedSearchIntegrator extendedIntegrator;
 
-	public void initialize(CompletableFuture<ExtendedSearchIntegrator> future) {
-		extendedIntegratorFuture = future;
+	public void initialize(ExtendedSearchIntegrator extendedIntegrator) {
+		this.extendedIntegrator = extendedIntegrator;
 	}
 
 	public ExtendedSearchIntegrator getSearchIntegrator() {
-		final CompletableFuture<ExtendedSearchIntegrator> value = this.extendedIntegratorFuture;
+		ExtendedSearchIntegrator value = this.extendedIntegrator;
 		if ( value != null ) {
-			return value.join();
+			return value;
```

Comment:
The `InitializingIntegratorState` is different as it is more likely triggered by our own integration points relying on ORM's listeners. This specific case however is meant to serve user threads, which should really not be free to run before we're initialized. In case that happens, I think having the error message is clearer than a deadlock to defend against "dodgy integration patterns"